### PR TITLE
[SIL-86] Add fetching pull requests test

### PR DESCRIPTION
**Pull request by kezc**
- Added tests for `PullRequestDataSource#getPullRequestsForUser`,
- changed `PullRequest#updatedAt` type to LocalDateTime so it matches the rest of our code

### CODE REVIEW

File: app/src/test/java/com/appunite/loudius/network/datasource/PullRequestsNetworkDataSourceTest.kt [hunk 1]
```diff
@@ -97,7 +258,8 @@ class PullRequestsNetworkDataSourceTest {
                     "exampleOwner",
                     "exampleRepo",
                     "exampleNumber",
-                )
+
```
Comment (kezc) [inline app/src/test/java/com/appunite/loudius/network/datasource/PullRequestsNetworkDataSourceTest.kt]: suggestion: unnecessary line break

File: app/src/main/java/com/appunite/loudius/ui/pullrequests/PullRequestsScreen.kt [hunk 2]
```diff
@@ -129,31 +130,31 @@ fun PullRequestsScreenPreview() {
                     number = 0,
                     repositoryUrl = "${Constants.BASE_API_URL}/repos/appunite/Stefan",
                     title = "[SIL-67] Details screen - network layer",
-                    updatedAt = "2021-11-29T16:31:41Z",
+                    updatedAt = ZonedDateTime.parse("2023-03-07T09:24:24Z").toLocalDateTime(),
```
Comment (Krzysiudan) [inline app/src/main/java/com/appunite/loudius/ui/pullrequests/PullRequestsScreen.kt]: **Suggestion :** You can make it shorter by :
`
LocalDateTime.parse("2023-03-07T09:24:24")
`
In other places too.

File: app/src/test/java/com/appunite/loudius/network/datasource/PullRequestsNetworkDataSourceTest.kt [hunk 3]
```diff
@@ -18,30 +20,186 @@ import org.junit.jupiter.api.Assertions
...
+                        ),
+                    ),
+                ),
+            )
+
+            assertEquals(expected, actualResponse) { "Data should be valid" }
+        }
+
+        @Test
+        fun `Given incorrect access token WHEN processing request THEN return failure with Network error`() =
```
Comment (Krzysiudan) [inline app/src/test/java/com/appunite/loudius/network/datasource/PullRequestsNetworkDataSourceTest.kt]: **Issue:** Name states it returns network error but test states it expects UnknownError. 
Please fix that in other tests in this file. I've made that mistake 😅

File: app/src/test/java/com/appunite/loudius/network/datasource/PullRequestsNetworkDataSourceTest.kt [hunk 4]
```diff
@@ -225,7 +384,7 @@ class PullRequestsNetworkDataSourceTest {
                             "1",
                             User(33498031, "exampleUser"),
                             ReviewState.COMMENTED,
-                            LocalDateTime.parse("2023-03-02T10:21:36"),
+                            ZonedDateTime.parse("2023-03-02T10:21:36Z").toLocalDateTime(),
```
Comment (Krzysiudan) [inline app/src/test/java/com/appunite/loudius/network/datasource/PullRequestsNetworkDataSourceTest.kt]: **Question:** Why did you change that? 
I don't think that we need to show in expected data that we are getting that from the `ZonedDateTime`. It's only about making correct data.

File: app/src/main/java/com/appunite/loudius/ui/pullrequests/PullRequestsScreen.kt (hunk 2 above)
Comment (kezc) [reply app/src/main/java/com/appunite/loudius/ui/pullrequests/PullRequestsScreen.kt]: I prefer to have the same format as API or have it in a human-readable way :| I don't like that we remove the 'z' and have yet another date format in the app

File: app/src/test/java/com/appunite/loudius/network/datasource/PullRequestsNetworkDataSourceTest.kt (hunk 3 above)
Comment (kezc) [reply app/src/test/java/com/appunite/loudius/network/datasource/PullRequestsNetworkDataSourceTest.kt]: we probably should map this error to some UnauthorizedError or something instead, because this is a known error :P WDYT?

File: app/src/main/java/com/appunite/loudius/ui/pullrequests/PullRequestsScreen.kt (hunk 2 above)
Comment (Krzysiudan) [reply app/src/main/java/com/appunite/loudius/ui/pullrequests/PullRequestsScreen.kt]: What do you mean by that? Do you recommend using ZonedDateTime all over the app? 
Here, after all, you have LocalDateTime anyway and it is without `Z`. So it's only a longer way of creating the same.

File: app/src/test/java/com/appunite/loudius/network/datasource/PullRequestsNetworkDataSourceTest.kt (hunk 3 above)
Comment (Krzysiudan) [reply app/src/test/java/com/appunite/loudius/network/datasource/PullRequestsNetworkDataSourceTest.kt]: Yeah, yeah definitely. But it's more like an error-handling topic, so the task is for later. But if you want to implement that now, then why not.

Comment (Krzysiudan) [reply app/src/test/java/com/appunite/loudius/network/datasource/PullRequestsNetworkDataSourceTest.kt]: You forgot to correct the name here.

**Review by Krzysiudan — APPROVED** [review]
Approved but one name wasn't corrected 😅
